.github/workflows/test.yml

```diff
@@ -0,0 +1,138 @@
+          echo "::set-output name=commit_ref::${{ github.event.client_payload.pull_request.head.sha }}"
+        elif [ "${{ steps.get_version.outputs.VERSION }}" != "" ]; then
+          echo "::set-output name=master::false"
+          echo "::set-output name=architectures::linux/amd64,linux/arm64"
+          echo "::set-output name=commit_ref::${{ steps.get_version.outputs.VERSION }}"
+        else
+          echo "::set-output name=master::true"
+          echo "::set-output name=architectures::linux/amd64,linux/arm64"
+          echo "::set-output name=commit_ref::${{ github.sha }}"
+        fi
+
+        # The suffix to append to the repository name if not triggered by a push for a release
+        [[ "${{ steps.get_version.outputs.VERSION }}" == "" ]] && \
+          echo "::set-output name=repo-suffix::-ci" ||
+          echo "::set-output name=repo-suffix::"
+
+        if [ "${{ github.event_name }}" != "repository_dispatch" ]; then
+           echo "::set-output name=ok-to-continue::true"
+           echo "::set-output name=repo-name::${{ github.repository }}"
+        elif [ "${{ github.event.client_payload.github.payload.repository.full_name }}" == "${{ github.repository }}" ]; then
+           echo "::set-output name=ok-to-continue::true"
+           echo "::set-output name=repo-name::${{ github.repository }}"
+        else
+           echo "::set-output name=ok-to-continue::false"
+           echo "::set-output name=repo-name::${{ github.event.client_payload.github.payload.repository.full_name }}"
+        fi
+
+
+  test:
+    name: Launch Test
+    runs-on: ubuntu-20.04
+    needs: configure    
+    steps:
+
+    - name: Set up Go 1.16
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+      id: go
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+      with:
+        ref: "${{ needs.configure.outputs.commit_ref }}"
+        repository: "${{ needs.configure.outputs.repo-name }}"
+        persist-credentials: false
+
+    - name: Cache Docker layers
+      uses: actions/cache@v2
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-${{ matrix.component }}-buildx-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.component }}-buildx-
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Build (and Publish) liqo-test image
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        platforms: linux/amd64
+        tags: |
+          liqo/liqo-test${{ needs.configure.outputs.repo-suffix }}:latest
+          liqo/liqo-test${{ needs.configure.outputs.repo-suffix }}:${{ needs.configure.outputs.commit_ref }}
+        push: true
```

Comment:
Could avoid to push, since it is used only as next step (and it requires also a discrete amount of time to upload/download given its size). Instead, it should be sufficient to load in docker (load: true)